storm-core/src/jvm/backtype/storm/scheduler/Cluster.java

```diff
@@ -90,6 +92,21 @@ public Cluster(INimbus nimbus, Map<String, SupervisorDetails> supervisors, Map<S
         }
         this.conf = storm_conf;
     }
+
+    /**
+     * Get a copy of this cluster object
+     */
+    public Cluster getCopy() {
+        HashMap<String, SchedulerAssignmentImpl> newAssignments = new HashMap<String, SchedulerAssignmentImpl>();
+        for (Map.Entry<String, SchedulerAssignmentImpl> entry : this.assignments.entrySet()) {
+            newAssignments.put(entry.getKey(), new SchedulerAssignmentImpl(entry.getValue().getTopologyId(), entry.getValue().getExecutorToSlot()));
+        }
+        Map newConf = new HashMap<String, Object>();
+        newConf.putAll(this.conf);
```

Comment:
also if you are using an IDE, typing this will bring up all of the available member variables and methods which makes it easier to look up stuff